src/Service/HealthCheck/HealthCheckService.php

```diff
@@ -50,8 +56,24 @@ public function getHealthCheckResult(): HealthCheckResult
 
         $cacheStatistics = $resultCacheImplementation->getStats();
 
+        $connection = $this->entityManager->getConnection();
+
+        try {
+            $testQuery = $connection->getDatabasePlatform()->getDummySelectSQL();
+            $testQueryResult = (bool)$connection->fetchOne($testQuery);
+        } catch (Exception $exception) {
+            $testQueryResult = $connection->isConnected();
+
+            $this->logger->critical(
+                sprintf(
+                    'DB connection unavailable. Got `%s` exception from DBAL',
+                    $exception->getMessage()
+                )
+            );
+        }
+
         return new HealthCheckResult(
-            $this->entityManager->getConnection()->ping(),
+            $testQueryResult,
             ($cacheStatistics['uptime'] ?? 0) > 0
         );
     }
```

Comment:
What do you think about this?

```
try {
            $connection = $this->entityManager->getConnection();
            $isDatabaseConnectionAvailable = (bool)$connection->fetchOne(
                $connection->getDatabasePlatform()->getDummySelectSQL()
            );
        } catch (Exception $exception) {
            $this->logger->critical(
                sprintf(
                    'DB connection unavailable. Got `%s` exception from DBAL',
                    $exception->getMessage()
                )
            );

            $isDatabaseConnectionAvailable = false;
        }

        return new HealthCheckResult(
            $isDatabaseConnectionAvailable,
            ($cacheStatistics['uptime'] ?? 0) > 0
        );
```

Reply:
It can also work. But @Skyld , do you see a big difference and profit in amending (or, ofc, not amending) a call to `$connection->isConnected()`?

Reply:
I just don't get the idea of invoking `isConnected` method in the catch clause. If we have an exception in `SELECT 1` query, the database connection is not ok, so I thought the check could be simplified.